Formula/skaa.rb

```diff
@@ -0,0 +1,44 @@
+class Skaa < Formula
+  desc "Seven Kingdoms: Ancient Adversaries"
+  homepage "https://7kfans.com"
+  url "https://github.com/the3dfxdude/7kaa/releases/download/v2.15.5/7kaa-2.15.5.tar.xz"
+  sha256 "350a2681985feb4b71d20677d1a6d11921b9e3d97facbc94e4f020e848d8ab2b"
+  license "GPL-2.0-only"
+
+  # we only use gcc for building the app
+  depends_on "gcc@11" => :build
```

Comment:
How do you handle linkage with GCC runtime libraries?

Reply:
I have to admit I am not well versed in C/C++ and this is the first time I'm making a Homebrew formula.

Clang has some sort of gcc-like front-end that has issues building the game. Basically Clang doesn't recognise the 387 unit properly in the configure script. However, I figured out I could build the game using gcc 11. Afterwards, I don't think gcc is needed. I would assume the game just uses the Clang runtime libraries. 

But I don't know if there's any way to check.  

Maybe to verify I should uninstall gcc and see if the game still runs?

Any advise here would be welcome.

Reply:
What is the output of `brew linkage skaa` after building? Doesn't the latest `gcc` work?

Reply:
For reference, it seems some games also check for the existence of a binary, an example:
- https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/rogue.rb

Other games might check a flag in the binary, would something like that be enough as well for skaa? 
- https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/brogue.rb
- https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/nsnake.rb
- https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/abuse.rb

Reply:
@carlocab 

```
System libraries:
  /System/Library/Frameworks/OpenAL.framework/Versions/A/OpenAL
  /usr/lib/libSystem.B.dylib
  /usr/lib/libcurl.4.dylib
  /usr/local/lib/libenet.7.dylib
Homebrew libraries:
  /usr/local/opt/gcc@11/lib/gcc/11/libstdc++.6.dylib (gcc@11)
  /usr/local/opt/sdl2/lib/libSDL2-2.0.0.dylib (sdl2)
Undeclared dependencies with linkage:
  gcc@11
Dependencies with no linkage:
  enet
```

I will verify if latest `gcc` works. Previously I believe I had issues, hence choosing an older version, but I should be 100% sure.

Reply:
Seems latest gcc doesn't work - it seems when trying to use the default gcc, the gcc front-end to Clang is used (in `/usr/bin/gcc`). And therefore the issue pops up that the 387-like processor is not properly recognised.

For most of the game this is not a problem, however multiplayer will not work in such cases. I'd like to be able to build the game with multiplayer support, hence using an explicit `gcc` version.

Reply:
The latest `gcc` should be invoked as `gcc-12`. `brew` will handle that on its own if you use `fails_with` as suggested. Please apply my suggestions so we can run those changes in CI.

Based on the output of `brew linkage` you will need to have the `gcc@*` used to build this installed. This is why we should use the latest if possible, because users are more likely to already have that installed.

Reply:
@carlocab I verified - `gcc-12` did not work, but `g++-12` works fine. Will update the PR very soon. Only trying to update the test.